.github/actions/docker/action.yaml

```diff
@@ -39,11 +40,23 @@ runs:
     # were changed we need to use an existing image (e.g. main).  In cases where docker image is rebuilt we have
     # to use tag generated by the image build.
     - name: Set Docker Tag
-      if: ${{ (steps.changed-files.outputs.any_changed == 'false') && (github.event_name == 'pull_request') }}
+      id: set-docker-tag-presubmit-non-fork
+      if: ${{ (steps.changed-files.outputs.any_changed == 'false') && (github.event_name == 'pull_request') && (github.repository == github.event.pull_request.head.repo.full_name) }}
```

Comment:
I believe you can just use the boolean `github.event.pull_request.head.repo.fork`